src/executables/unload_schema.c

```diff
@@ -815,6 +828,163 @@ export_serial (print_output & output_ctx)
+  DB_QUERY_ERROR query_error;
+  DB_VALUE values[SYNONYM_VALUE_INDEX_MAX];
+  const char *synonym_name = NULL;
+  DB_OBJECT *synonym_owner = NULL;
+  const char *synonym_owner_name = NULL;
+  int is_public = 0;
+  const char *target_name = NULL;
+  const char *target_owner_name = NULL;
+  const char *comment = NULL;
+  bool is_dba_group_member = false;
+  int i = 0;
+  int save = 0;
+  int error = NO_ERROR;
+
+  // *INDENT-OFF*
+  const char *query = "SELECT [name], "
+			     "[owner], "
+			     "[owner].[name], "
+			     "[is_public], "
+			     "[target_name], "
+			     "[target_owner].[name], "
+			     "[comment] "
+			"FROM [_db_synonym]";
+  // *INDENT-ON*
+
+  query_error.err_lineno = 0;
+  query_error.err_posno = 0;
+
+  AU_DISABLE (save);
+
+  error = db_compile_and_execute_local (query, &query_result, &query_error);
+  if (error < 0)
+    {
+      ASSERT_ERROR ();
+      goto end;
+    }
+
+  is_dba_group_member = au_is_dba_group_member (Au_user);
+
+  output_ctx ("\n\n");
+
+  while (db_query_next_tuple (query_result) == DB_CURSOR_SUCCESS)
+    {
+      for (i = 0; i < SYNONYM_VALUE_INDEX_MAX; i++)
+	{
+	  error = db_query_get_tuple_value (query_result, i, &values[i]);
+	  if (error != NO_ERROR)
+	    {
+	      ASSERT_ERROR ();
+	      goto end;
+	    }
+
+	  /* Validation of the result value */
+	  switch (i)
+	    {
+	    case SYNONYM_NAME:
+	    case SYNONYM_OWNER_NAME:
+	    case SYNONYM_TARGET_NAME:
+	    case SYNONYM_TARGET_OWNER_NAME:
+	      {
+		if (DB_IS_NULL (&values[i]) || DB_VALUE_TYPE (&values[i]) != DB_TYPE_STRING)
+		  {
+		    ERROR_SET_ERROR (error, ER_SYNONYM_INVALID_VALUE);
+		    goto end;
+		  }
+	      }
+	      break;
+
+	    case SYNONYM_OWNER:
+	      {
+		if (DB_IS_NULL (&values[i]) || DB_VALUE_TYPE (&values[i]) != DB_TYPE_OBJECT)
+		  {
+		    ERROR_SET_ERROR (error, ER_SYNONYM_INVALID_VALUE);
+		    goto end;
+		  }
+	      }
+	      break;
+
+	    case SYNONYM_IS_PUBLIC:
+	      {
+		if (DB_IS_NULL (&values[i]) || DB_VALUE_TYPE (&values[i]) != DB_TYPE_INTEGER)
+		  {
+		    ERROR_SET_ERROR (error, ER_SYNONYM_INVALID_VALUE);
+		    goto end;
+		  }
+	      }
+	      break;
+
+	    case SYNONYM_COMMENT:
+	      {
+		if (DB_IS_NULL (&values[i]) == false && DB_VALUE_TYPE (&values[i]) != DB_TYPE_STRING)
+		  {
+		    ERROR_SET_ERROR (error, ER_SYNONYM_INVALID_VALUE);
+		    goto end;
+		  }
+	      }
+	      break;
+
+	    default:
+	      ERROR_SET_ERROR (error, ER_SYNONYM_INVALID_VALUE);
+	      goto end;
+	    }
+	}
+
+      synonym_name = db_get_string (&values[SYNONYM_NAME]);
+      synonym_owner = db_get_object (&values[SYNONYM_OWNER]);
+      synonym_owner_name = db_get_string (&values[SYNONYM_OWNER_NAME]);
+      is_public = db_get_int (&values[SYNONYM_IS_PUBLIC]);
+      target_name = db_get_string (&values[SYNONYM_TARGET_NAME]);
+      target_owner_name = db_get_string (&values[SYNONYM_TARGET_OWNER_NAME]);
+
+      if (!is_dba_group_member && !ws_is_same_object (Au_user, synonym_owner))
```

Comment:
This check and continuing can be done earlier in the while loop, that is, before the for-loop.
It will reduce the execution time if db_query_get_tuple_value() is an expensive operation.

Reply:
Thank you. I agree with that.

However, unloading a schema rather than an object is not a big load,
so I will invest more in code brevity.